src/main/java/org/reactome/release/qa/diagram/DiagramDiseaseColorCheck.java

```diff
@@ -0,0 +1,107 @@
+package org.reactome.release.qa.diagram;
+
+import java.util.Collection;
+import java.util.List;
+
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.DiagramGKBReader;
+import org.gk.render.DefaultRenderConstants;
+import org.gk.render.Renderable;
+import org.gk.render.RenderablePathway;
+import org.reactome.release.qa.annotations.DiagramQACheck;
+import org.reactome.release.qa.common.QACheckerHelper;
+import org.reactome.release.qa.common.QAReport;
+
+/**
+ * QA check to flag any oddities in disease reaction coloring.
+ * The reaction edges of an RLE with a disease tag should be red.
+ * Likewise if an RLE is not tagged with disease, reaction edges should be black.
+ * Disease-tagged entities and/or entities with non-human species (or a relatedSpecies attribute) should colored red.
+ * RLEs QA consideration: manually applied colors may differ. RGB comparison to make sure same color is applied.
+ * Have predefined color for certain entities.
+ */
+@DiagramQACheck
+public class DiagramDiseaseColorCheck extends AbstractDiagramQACheck {
+
+    public DiagramDiseaseColorCheck() {
+    }
+
+    @Override
+    public QAReport executeQACheck() throws Exception {
+        QAReport report = new QAReport();
+        Collection<GKInstance> pathwayDiagrams = getPathwayDiagrams();
+        DiagramGKBReader reader = new DiagramGKBReader();
+        for (GKInstance diagram : pathwayDiagrams) {
+            checkPathwayDiagram(diagram, reader, report);
+        }
+        report.setColumnHeaders("PathwayDiagram_DBID",
+                                "Pathway_DisplayName",
+                                "Pathway_DBID",
+                                "Renderable_DisplayName",
+                                "Renderable_DBID",
+                                "MostRecentAuthor");
+        return report;
+    }
+
+    private void checkPathwayDiagram(GKInstance diagram, DiagramGKBReader reader, QAReport report) throws Exception {
+        RenderablePathway pathway = reader.openDiagram(diagram);
+        String modDate = QACheckerHelper.getLastModificationAuthor(diagram);
```

Comment:
Move this statement before the following for loop (line 54) to avoid calling it in case nothing in the diagram.